Framework/HistogramData/src/Histogram.cpp

```diff
@@ -253,5 +253,21 @@ template <> void Histogram::checkSize(const BinEdges &binEdges) const {
     throw std::logic_error("Histogram: size mismatch of BinEdges\n");
 }
 
+/** Resets the size of the internal x, dx, y, and e data structures
+
+  The argument refers to the requested y length. All other lengths are
+  then inferred automatically.*/
```

Comment:
Maybe expand the doxygen to mention that the behavior is like `std::vector::resize` (truncating, or appending zeros).